labgrid/driver/sshdriver.py

```diff
@@ -111,12 +110,19 @@ def _start_own_master_once(self, timeout):
             ]
 
         env = os.environ.copy()
+        pass_file = ''
         if self.networkservice.password:
-            env['SSHPASS'] = self.networkservice.password
+            fd, pass_file = tempfile.mkstemp()
+            os.fchmod(fd, stat.S_IRWXU)
+            env = {'SSH_ASKPASS': pass_file, 'DISPLAY':'0:0'}
```

Comment:
Newer SSH versions also have SSH_ASKPASS_REQUIRE, which we could set to `force`. Maybe we could set DISPLAY to a clearly invalid value without breaking this mechanism?

Reply:
I have tried openssh 8.4p1 nightly and with SSH_ASKPASS_REQUIRE=force it works without DISPLAY variable and without the preexec_fn=os.setsid.